apps/labs/posts/scikit-lego-narwhals.md

```diff
@@ -0,0 +1,115 @@
+---
+title: 'How Narwhals and scikit-lego came together to achieve dataframe-agnosticism'
+authors: [marco-gorelli]
+published: May 26, 2024
+description: 'And how your library can become dataframe-agnostic too'
+category: [PyData ecosystem]
+featuredImage:
+  src: /posts/scikit-lego-narwhals/scikit_lego_narwhals_handshake.png
+  alt: 'Scikit-lego and Narwhals logos shaking hands'
+hero:
+  imageSrc: /posts/scikit-lego-narwhals/hero.png
+  imageAlt: 'Scikit-lego and Narwhals logos'
+---
+
+[Scikit-lego](https://github.com/koaning/scikit-lego) is a Python project from the [scikit-learn](https://scikit-learn.org/stable/) ecosystem that contributes extra estimators for machine-learning pipelines.
+It's a relatively popular project with over 1000 Github stars and 20,000 downloads a month.
+It's popularity comes from industry as many of the tools that it provides aren't proven to be state-of-the-art,
+but fall a bit more in the 'tricks that work' category. These features include a GroupedEstimator which
+can effectively run full pipelines split per subset, a TimeGapSplit cross validator for timeseries
+that adds a gap between train and test set, and a whole suite of techniques that leverage mixture methods
+for outlier detection and non-linear classification.
+
+Recently, because polars has been gaining traction, the library maintainers have been looking for a simple way to support multiple dataframe implementations. The project still wants to support pandas, but it would be a shame if newer dataframes couldn't be supported.
+
+Enter narwhals. As of scikit-lego 0.9.0, you can also pass dataframes from Polars, Modin, cuDF, and in principle a whole host
+of other dataframes.
+But, how does it work? Why did they use Narwhals?
+Why don't they just
+convert to pandas internally on the user's behalf?
```

Comment:
What is going on with the spacing here

Reply:
they render as a single paragraph, which is what i was going for: https://labs-68vdxlovu-quansight.vercel.app/blog/scikit-lego-narwhals

I'd probably just moved things around

just for my understand, does it make a difference?

Reply:
No, I was just double checking